WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteCardAndItem.kt

```diff
@@ -85,27 +84,35 @@ sealed class MySiteCardAndItem(open val type: Type, open val activeQuickStartIte
         }
 
         sealed class PostCard(
-            open val postCardType: PostCardType
-        ) : Card(if (postCardType == CREATE_FIRST) POST_CARD_CREATE_FIRST else POST_CARD_DRAFT_OR_SCHEDULED) {
-            data class PostCardCreateFirst(
-                override val postCardType: PostCardType,
+            override val type: Type,
+            open val footerLink: FooterLink
+        ) : Card(type) {
+            data class PostCardWithoutPostItems(
+                val postCardType: PostCardType,
                 val title: UiString,
                 val excerpt: UiString,
-                @DrawableRes val imageRes: Int
-            ) : PostCard(postCardType = postCardType)
+                @DrawableRes val imageRes: Int,
+                override val footerLink: FooterLink
+            ) : PostCard(type = POST_CARD_WITHOUT_POST_ITEMS, footerLink = footerLink)
 
-            data class PostCardDraftOrScheduled(
-                override val postCardType: PostCardType,
+            data class PostCardWithPostItems(
+                val postCardType: PostCardType,
                 val title: UiString,
-                val postItems: List<PostItem>
-            ) : PostCard(postCardType = postCardType) {
+                val postItems: List<PostItem>,
+                override val footerLink: FooterLink
+            ) : PostCard(type = POST_CARD_WITH_POST_ITEMS, footerLink = footerLink) {
                 data class PostItem(
                     val title: UiString,
                     val excerpt: UiString?,
                     val featuredImageUrl: String?,
-                    val isTimeIconVisible: Boolean
+                    val isTimeIconVisible: Boolean = false
                 )
             }
+
+            data class FooterLink(
+                val label: UiString,
+                val onClick: ((postCardType: PostCardType) -> Unit)? = null
```

Comment:
Minor (🔍): Consider making the `onClick` non null, as if: `(postCardType: PostCardType) -> Unit`

I can only see the test benefiting from it being nullable and with a default `null` value, but this shouldn't drive the main code as the `FooterLink` should never be allowed to be created with such a state, right?

Reply:
In c19ce78

Reply:
Great, many thanks @ashiagr !

Btw, wouldn't `FooterLink(label = XYZ) { }` do the job as well, without the need of an explicit `mock`? 🤔

Reply:
It can do, again, a matter of preference I think.

Reply:
No need to do, I was don't asking because maybe there was some problem with using a lambda on the tests, sometime lambdas are painful to test and I wasn't sure if that was the case here.